litellm/integrations/literal_ai.py

```diff
@@ -0,0 +1,193 @@
+#### What this does ####
+# This file contains the LiteralAILogger class which is used to log steps to the LiteralAI observability platform.
+import copy
+import traceback
+
+
+import litellm
+from litellm._logging import verbose_logger
+from litellm.litellm_core_utils.redact_messages import redact_user_api_key_info
+
+
+class LiteralAILogger:
+    def __init__(
+        self,
+        literalai_api_key=None,
+        literalai_api_url=None,
+    ):
+        try:
+            from literalai import LiteralClient
+        except Exception as e:
+            raise Exception(
+                f"\033[91mLiteralAI not installed, try running 'pip install literalai' to fix this error: {e}\n{traceback.format_exc()}\033[0m"
+            )
+    
+        self.client = LiteralClient(api_key=literalai_api_key, url=literalai_api_url)
+
+       
+    @staticmethod
+    def add_metadata_from_header(litellm_params: dict, metadata: dict) -> dict:
```

Comment:
This shouldn't be necessary if you're using our standard logging payload - https://github.com/BerriAI/litellm/blob/ab153c3e8dab6be093eba3726ada19da176ee8cc/litellm/integrations/gcs_bucket.py#L63

It should guarantee the metadata, etc. is correctly handled for all the different llm api calls (audio, embedding, chat, etc.) as well as if users set flags for masking certain information.